### ISSUE TITLE
Docs / Dev Experience for Infra Providers

### ISSUE THREAD

**Issue by bmgentile** [hiero-ledger/hiero-docs#460]
Hey folks — we’re running into a developer experience and docs pain point that has come up via the Validation Cloud integration that I think we can look at as a challenge to solve together — it’s something new for us.

**Background**
- There are web3 partner integrators / service providers with a “job-to-be-done” of integrating things like: JSON-RPC, The Graph, Mirror Node, etc. to provide to their users
- These providers are critical for driving the network effects needed for Hedera’s expansion across web3; folks like Arkhia, Validation Cloud, Alchemy (soon), and (hopefully) many others coming down the line
- I think we have done an incredible job as a team creating a positive developer experience / docs / etc. for end users of Hedera’s (and Swirlds Labs’) services (dapp developers)
- We are at a point in time where, as we desire to scale Hedera’s offerings to the rest of web3 via these providers, we need to bring that same stellar developer experience to those who are integrating and providing these services as an offering to developers.
- Validation Cloud has been running into issues around comprehension, implementation, and upgrade processes and it highlights a pain point that I anticipate we will see by others as we continue on this journey.

**Proposal**
- HashIO is a Swirlds Labs hosted version of JSON-RPC — it’s going to continue to be a productized offering / community service for developers and likely expand into including The Graph, Mirror Node, and other offerings that are open sourced via Hedera (Hedera repo)
- How we implemented HashIO and provide it as an offering to the Hedera developer community is information that is needed by these providers in a very easily consumable way.
- If we had http://docs.swirldslabs.com (or http://docs.hashio.com) that has an emphasis on “How to implement and maintain these services as an infrastructure provider for your customers” we will drastically:
    - Reduce time-to-market for service providers
    - Improve their ability/process for upgrading to the latest version(s) of these services in a timely manner
    - Reduce the amount of time SL engineering / DAs spend helping these service providers get up-to-speed
    - Improve the experience for customers/dapp developers of these service providers

**Comment by SimiHunjan** [comment]
hey @bmgentile ! Are you suggesting for Swirlds Labs to host their docs site for the services they provide?

**Comment by bmgentile** [comment]
Hey @SimiHunjan — no, Validation Cloud (and any other commercial provider of services that utilize the JSON-RPC codebase, The Graph, etc.) should be hosting their own end-user documentation specific to their services.

This proposal is for creating a docs.swirldslabs.com, which focuses on the creation of documentation that teaches commercial infrastructure providers _**how**_  Swirlds Labs has integrated these open source codebases into HashIO and how they, too, can implement these open source codebases into their commercial offerings.

The proposal is to meant to deliver an improved developer experience for companies like Validation Cloud, Alchemy, Arkhia, etc., where they need support on implementation of these open source codebases, in order to best serve their users.

The Validation Cloud citation is only an example that showcases how the "persona" of commercial infrastructure providers is one which we should have a focus on, because they are providing a service that brings network effects in adoption to the Hedera ecosystem.

**Comment by SimiHunjan** [comment]
yes sorry didn't mean for us to host all the service providers documentation I just thought you meant to have docs.swirldslabs.com for documentation on HashIO or HashScan.

Do you think we have enough content to host a SL docs site today? I was also wondering if Swirlds Labs should release a blog on how HashIO was integrated. I have no opinion just curious on your thoughts.

**Comment by bmgentile** [comment]
Ah, sorry — yes, I think we do have enough to host an SL docs website; believe there's a gap for "commercial service providers" (and even some devs who want to run their own infra), who are attempting to stand up the open source hedera explorer, JSON-RPC relay codebase, the graph, local node, et al, in the same way that we have/continue to do as Swirlds Labs "community service" offerings.

A big part of this is creating a really stellar developer experience (as we've done with Hedera docs) and applying it to this new persona; I view this this new persona as incredibly important for Hedera to be successful, especially as we scale.. this persona / provider is going to drive the network effects we need on the adoption front.
